torch/csrc/jit/passes/onnx/function_extraction.cpp

```diff
@@ -1020,6 +1071,28 @@ NodeAttrNameMap FunctionExtractor::run() {
   return node_attr_to_name;
 }
 
+Node* GetNodeOfMostRecentScope(Node* forward_node) {
+  // This function is used to retrieve the node representing the most recent
+  // ScopePtr. This function should only be invoked from module forward hook. At
+  // this point, module forward call is completed, and the most recent ScopePtr
+  // is popped from TracingState.
+  // This function inspects the node, and its subblock, to find
+  // the node associated with the most recent ScopePtr.
+
+  // Skip the "real" last node which is `return_node`.
+  TORCH_INTERNAL_ASSERT(forward_node->kind() == prim::TracedModuleForward);
+  auto* block = forward_node->blocks()[0];
+  for (auto* node : block->nodes().reverse()) {
+    if (node->kind() == prim::TracedModuleForward) {
+      auto* target_node = GetNodeOfMostRecentScope(node);
```

Comment:
In general, if the type name is pretty short and simple, I think it's a bit more readable to use it rather than `auto`.
In this case I think `Node*` is preferrable to `auto*`.

Reply:
I agree, especially for some ambiguous function returns that is not obvious about return type. For this particular case though, both function and variable name should reflect the type is `Node`, I feel it should be ok with `auto`. Let me know if you still prefer `Node*` here.